src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DraftReferralService.kt

```diff
@@ -389,7 +389,7 @@ class DraftReferralService(
     communityAPIReferralService.send(referral)
 
     val sentReferral = referralRepository.save(referral)
-    createReferralLocation(draftReferral)
+    createReferralLocation(draftReferral, sentReferral)
```

Comment:
Not really an issue, but you could also do a single `referralRepository.save(referral)` if you put this line above and don't save inside the method (I think?). Alternatively we could just check that the whole thing is rolled back if the referral table was saved and then the location save failed? I guess both approaches would need to be tested

Reply:
I also thought it would be good to do a single save on the referralRepository but we need to save the referral first in order to map ReferralLocation (It's being done this way for SupplierAssessment a couple lines below).

Reply:
ah okay. Fair enough 👍